packages/http-client/src/index.ts

```diff
@@ -13,19 +13,68 @@ export const loadProject = async ({
   if (apiUrl === undefined) {
     throw new Error("Webstudio API URL is required.");
   }
-  const baseUrl = new URL(`${apiUrl}`);
+  const urls = [];
+  const pages: Record<string, unknown> | null = {};
+
+  const baseUrl = new URL(apiUrl);
   const treeUrl = new URL(`/rest/tree/${projectId}`, baseUrl);
   const propsUrl = new URL(`/rest/props/${projectId}`, baseUrl);
   const breakpointsUrl = new URL(`/rest/breakpoints/${projectId}`, baseUrl);
+  const pagesUrl = new URL(`/rest/pages/${projectId}`, baseUrl);
 
-  const [tree, props, breakpoints] = await Promise.all([
+  const [tree, props, breakpoints, listOfPages] = await Promise.all([
     include.tree && (await fetch(treeUrl)).json(),
     include.props && (await fetch(propsUrl)).json(),
     include.breakpoints && (await fetch(breakpointsUrl)).json(),
+    include.pages && (await fetch(pagesUrl)).json(),
   ]);
+
+  if (include.pages && listOfPages.pages && listOfPages.pages.length > 0) {
+    for (const page of listOfPages.pages) {
+      urls.push({
+        url: new URL(`/rest/tree/${projectId}/${page.id}`, baseUrl),
+        path: page.path,
+        type: "tree",
+      });
+      urls.push({
+        url: new URL(`/rest/props/${projectId}/${page.id}`, baseUrl),
+        path: page.path,
+        type: "props",
+      });
+    }
+    if (urls.length === 0) {
+      return {
+        tree,
+        props,
+        breakpoints,
+        pages,
+      };
+    }
+
+    const pagesData = await Promise.all([
+      ...urls.map(async (url) => {
```

Comment:
.map returns a new array anyways
```suggestion
    const pagesData = await Promise.all(
      urls.map(async (url) => {
```